draft-packages/form/KaizenDraft/Form/Primitives/InputSearch/InputSearch.tsx

```diff
@@ -0,0 +1,104 @@
+import { Icon } from "@kaizen/component-library"
+import { LoadingSpinner } from "@kaizen/draft-loading-spinner"
+import classnames from "classnames"
+import React, { useState } from "react"
+import search from "@kaizen/component-library/icons/search.icon.svg"
+import clear from "@kaizen/component-library/icons/clear.icon.svg"
+
+import styles from "./styles.scss"
+
+export interface InputSearchProps
+  extends Omit<React.InputHTMLAttributes<HTMLInputElement>, "className"> {
+  classNameAndIHaveSpokenToDST?: string
+  inputRef?: React.RefObject<HTMLInputElement>
+  reversed?: boolean
+  loading?: boolean
+  primary?: boolean
+  secondary?: boolean
+  startIconAdornment?: React.ReactNode
+  endIconAdornment?: React.ReactNode
+}
+
+const InputSearch: React.FunctionComponent<InputSearchProps> = (
+  props: InputSearchProps
+) => {
+  const {
+    classNameAndIHaveSpokenToDST,
+    placeholder,
+    disabled = false,
+    defaultValue,
+    inputRef,
+    reversed = false,
+    loading,
+    startIconAdornment,
+    endIconAdornment,
+    primary,
+    secondary,
+    ...genericInputProps
+  } = props
+
+  const [value, setValue] = useState("")
```

Comment:
I wouldn't have thought this would be a stateful component 🤔 

I can see that we can supply an `onChange` and `value` as props that will overwrite the ones specified such as `setValue`, but what about the onClear?

Reply:
I think the idea is we have both options but @dougmacknz might weight in on this. 

Good catch I haven't implemented the `onClear` overwrite. Will do now 👍

Reply:
Guess I'm also trying to work out when I wouldn't want to not know what the value is? We are just setting the internal state here and not telling the consumer anything about it 🤷‍♀️ What purpose is this serving?

Reply:
We offer both options for our form inputs to allow the option of doing native browser input (though it sounds like that's invalid here because of the `onClear` need) and using the value via form submission.

Reply:
Interesting. I would have thought that you could achieve that behaviour by not supplying a `value` prop since inputs are inherently uncontrolled.

Reply:
This needs a bit of work, `value` is a prop but it's being ignored and defaulted to an empty string here, and the clearing functionality doesn't work for uncontrolled. Let's pair.

Reply:
> I would have thought that you could achieve that behaviour by not supplying a value prop since inputs are inherently uncontrolled.

You're right, except this component is a bit more tricky because it has the 'clear' functionality, and we're creating our own button and hiding the native one (which only shows in some browsers). Anyway, we have ended up abandoning the idea of uncontrolled on this component now because it complicates things a fair bit, and as you mentioned - it's probably unlikely that someone will need it.